pkg/cluster/arooperator.go

```diff
@@ -5,27 +5,34 @@ package cluster
 
 import (
 	"context"
-
-	"github.com/Azure/ARO-RP/pkg/operator/deploy"
 )
 
+func (m *manager) isIngressProfileAvailable() bool {
+	// We try to aqcuire the IngressProfiles data at frontend best effort enrichment time only.
+	// When we start deallocated VMs and wait for the API do become available again, we don't pick
+	// the information up, even though it would be available.
+	return len(m.doc.OpenShiftCluster.Properties.IngressProfiles) != 0
+}
+
 func (m *manager) ensureAROOperator(ctx context.Context) error {
-	dep, err := deploy.New(m.log, m.env, m.doc.OpenShiftCluster, m.arocli, m.extensionscli, m.kubernetescli)
-	if err != nil {
-		m.log.Errorf("cannot ensureAROOperator.New: %s", err.Error())
-		return err
+	//ensure the IngressProfile information is available from the cluster which is not the case when the cluster vms were freshly restarted.
+	if !m.isIngressProfileAvailable() {
```

Comment:
Just a reminder to update `createOrUpdateRouterIPFromCluster` once we merge this PR.